pkg/odo/cli/dev/dev.go

```diff
@@ -233,6 +233,10 @@ func (o *DevOptions) Run(ctx context.Context) error {
 	}()
 
 	portsBuf.Wait()
+	err = o.clientset.StateClient.SetForwardedPorts(portsBuf.GetForwaredPorts())
+	if err != nil {
+		fmt.Printf("unable to save forwarded ports to state file: %v", err)
```

Comment:
Shouldn't we exit right away if we cannot write to the state file? As I see it from the issue description, not exiting when such errors occur might create issues for other commands like `odo describe component`.
Of if an error here is not problematic, maybe we could print a warning instead (using `log.Warningf`).